Signal/src/ViewControllers/GroupViewHelper.m

```diff
@@ -0,0 +1,204 @@
+                    [self takePicture];
+                }];
+    [actionSheetController addAction:takePictureAction];
+
+    UIAlertAction *choosePictureAction = [UIAlertAction
+        actionWithTitle:NSLocalizedString(@"MEDIA_FROM_LIBRARY_BUTTON", @"media picker option to choose from library")
+                  style:UIAlertActionStyleDefault
+                handler:^(UIAlertAction *_Nonnull action) {
+                    [self chooseFromLibrary];
+                }];
+    [actionSheetController addAction:choosePictureAction];
+
+    [self.delegate.fromViewController presentViewController:actionSheetController animated:true completion:nil];
+}
+
+- (void)takePicture
+{
+    OWSAssert([NSThread isMainThread]);
+    OWSAssert(self.delegate);
+
+    UIImagePickerController *picker = [[UIImagePickerController alloc] init];
+    picker.delegate = self;
+    picker.allowsEditing = NO;
+    picker.sourceType = UIImagePickerControllerSourceTypeCamera;
+
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        picker.mediaTypes = [[NSArray alloc] initWithObjects:(NSString *)kUTTypeImage, nil];
+        [self.delegate.fromViewController presentViewController:picker
+                                                       animated:YES
+                                                     completion:[UIUtil modalCompletionBlock]];
+    }
+}
+
+- (void)chooseFromLibrary
+{
+    OWSAssert([NSThread isMainThread]);
+    OWSAssert(self.delegate);
+
+    UIImagePickerController *picker = [[UIImagePickerController alloc] init];
+    picker.delegate = self;
+    picker.sourceType = UIImagePickerControllerSourceTypeSavedPhotosAlbum;
+
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeSavedPhotosAlbum]) {
+        picker.mediaTypes = [[NSArray alloc] initWithObjects:(NSString *)kUTTypeImage, nil];
+        [self.delegate.fromViewController presentViewController:picker
+                                                       animated:YES
+                                                     completion:[UIUtil modalCompletionBlock]];
+    }
+}
+
+/*
+ *  Dismissing UIImagePickerController
+ */
+
+- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker
+{
+    OWSAssert([NSThread isMainThread]);
+    OWSAssert(self.delegate);
+
+    [self.delegate.fromViewController dismissViewControllerAnimated:YES completion:nil];
+}
+
+/*
+ *  Fetch data from UIImagePickerController
+ */
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info
+{
+    OWSAssert([NSThread isMainThread]);
+    OWSAssert(self.delegate);
+
+    UIImage *rawAvatar = [info objectForKey:UIImagePickerControllerOriginalImage];
+
+    if (rawAvatar) {
+        // TODO: There may be a bug here.
+        UIImage *resizedAvatar = [rawAvatar resizedImageToFitInSize:CGSizeMake(100.00, 100.00) scaleIfSmaller:NO];
```

Comment:
Does this need to be rebased against the recent group avatar changes? Expecting to see something like 210px IIRC.